Alicerce/Sources/Persistence/DiskMemoryPersistence.swift

```diff
@@ -0,0 +1,326 @@
+
+        if let data = cachedData(for: key) {
+            return completion { data }
+        }
+
+        // TODO: Check for object on disk and cache it if don't available
+
+        diskData(for: key) { [weak self] inner in
+            do {
+                let data = try inner()
+
+                self?.setCachedData(data, for: key)
+
+                completion { data }
+            } catch {
+                completion { throw error }
+            }
+        }
+    }
+
+    public func setObject(_ object: Data, for key: Persistence.Key, completion: @escaping CompletionClosure<Void>) {
+
+        setCachedData(object, for: key)
+
+        setDiskData(object, for: key, completion: completion)
+    }
+
+    public func removeObject(for key: Persistence.Key, completion: @escaping CompletionClosure<Void>) {
+
+        removeCachedData(for: key)
+
+        removeDiskData(for: key, completion: completion)
+    }
+
+    // MARK: - Private Methods
+
+    // MARK: - NSCache Related Operations
+
+    private func cachedData(for key: Persistence.Key) -> Data? {
+        return cache.object(forKey: key.nsString) as Data?
+    }
+
+    private func setCachedData(_ data: Data, for key: Persistence.Key) {
+        cache.setObject(data.nsData, forKey: key.nsString)
+    }
+
+    private func removeCachedData(for key: Persistence.Key) {
+        cache.removeObject(forKey: key.nsString)
+    }
+
+    // MARK: - Disk Related Operations
+
+    private func diskData(for key: Persistence.Key, completion: @escaping CompletionClosure<Data>) {
+        guard diskCacheEnabled else { return completion { throw PersistenceError.other(Error.diskCacheDisabled) } }
+
+        let readOperation = DiskMemoryBlockOperation() { [unowned self] in
+            let path = self.diskPath(for: key)
+
+            guard let fileData = self.fileManager.contents(atPath: path) else {
+                return completion { throw PersistenceError.other(Error.noDataForKey) }
+            }
+
+            return completion { fileData }
+        }
+
+        readOperationQueue.addOperation(readOperation)
+    }
+
+    private func setDiskData(_ data: Data, for key: Persistence.Key, completion: @escaping CompletionClosure<Void>) {
+        guard diskCacheEnabled else { return completion { throw PersistenceError.other(Error.diskCacheDisabled) } }
+
+        let writeOperation = DiskMemoryBlockOperation() { [unowned self] in
+            let path = self.diskPath(for: key)
+
+            var isDir = ObjCBool(false)
+            let fileExists = self.fileManager.fileExists(atPath: path, isDirectory: &isDir)
+
+            guard isDir.boolValue == false else {
+                return completion { throw PersistenceError.other(Error.fileNotCreated) }
+            }
+
+            guard self.fileManager.createFile(atPath: path, contents: data) else {
+                return completion { throw PersistenceError.other(Error.fileNotCreated) }
+            }
+
+            if !fileExists {
```

Comment:
Please be consistent regarding `false` comparisons 😛  (we appear to be using `== false` 😆)